.github/workflows/publish_charm.yaml

```diff
@@ -0,0 +1,13 @@
+name: Publish to edge
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
I must be a little confused... are tests run before the charm is published to the edge channel?

Reply:
The tests are run on the PR, once the tests pass, this workflow runs when the PR is merged with main to push the changes to the `edge` channel on Charmhub.